src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -192,23 +236,81 @@ public void setNumberOfInstances(int numberOfInstances) {
 
 			List<String> urls = sas.getUrls();
 			if (urls != null && !urls.isEmpty()) {
-				v.setAccessURL(this.determineAccessURL(v.getTarget().getProtocol(), urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns(), v.getTarget().getPath()));
+				// Set the access url to the selected route (without any protocol or path yet)
+				v.setAccessURL(this.determineApplicationRoute(urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns()));				
+      }      
+           
+			v.setNumberOfInstances(sas.getInstances());
+			
+			return Mono.just(v);
+    });
+        
+    Flux<List<RouteResource>> routeApplicationFlux = osaVectorSpaceFlux.flatMapSequential(v -> this.getAppRoutes(v.getApplicationId()));
+		Flux<OSAVector> osaVectorApplicationRouteFlux = Flux.zip(osaVectorApplicationFlux, routeApplicationFlux).flatMap(tuple -> {
+			OSAVector v = tuple.getT1();
+			
+			if (INVALID_ROUTES == tuple.getT2()) {
```

Comment:
I think this is my lack of familiarity again. I changed it to check for an empty list rather than a null list.

If the list is empty (ie there are no routes) then the target will be dropped.